compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -171,45 +178,91 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                     | Res::Local(_) => true,
                     _ => false,
                 },
-            )
+                suggestion: None,
+            }
         } else {
             let item_span = path.last().unwrap().ident.span;
-            let (mod_prefix, mod_str) = if path.len() == 1 {
-                (String::new(), "this scope".to_string())
+            let (mod_prefix, mod_str, suggestion) = if path.len() == 1 {
+                debug!(
+                    "self.diagnostic_metadata.current_impl_items={:?}",
+                    self.diagnostic_metadata.current_impl_items
+                );
+                debug!(
+                    "self.diagnostic_metadata.current_function={:?}",
+                    self.diagnostic_metadata.current_function
+                );
+                let suggestion = if let Some(items) = self.diagnostic_metadata.current_impl_items
+                    && let Some((fn_kind, _)) = self.diagnostic_metadata.current_function
+                    && self.current_trait_ref.is_none()
+                    && let Some(FnCtxt::Assoc(_)) = fn_kind.ctxt()
+                    && let Some(item) = items.iter().find(|i| {
+                        if let AssocItemKind::Fn(fn_) = &i.kind
+                            && fn_.sig.decl.inputs.iter().all(|i| match &i.ty.kind {
+                                TyKind::Rptr(_, ty) => match ty.ty.kind {
+                                    TyKind::ImplicitSelf => false,
+                                    _ => true,
+                                }
+                                TyKind::ImplicitSelf => false,
+                                _ => true
+                            })
+                            && i.ident.name == item_str.name
+                        {
+                            debug!("name={}", item_str.name);
+                            debug!("fn_.sig.decl.inputs={:?}", fn_.sig.decl.inputs);
```

Comment:
```suggestion
                            debug!(?item_str.name);
                            debug!(?fn_.sig.decl.inputs);
```
or completely remove these debug statements '^^